### PR TITLE
Feature/tt 1012 delete unsaved ps

### PR DESCRIPTION
**Pull request by sarahentzel**
also includes change to question when leaving unsaved sections and passages

### CODE REVIEW

File: src/components/PlanSheet.tsx [hunk 1]
```diff
@@ -85,7 +85,7 @@ interface IProps extends IStateProps {
   setChanged?: (v: boolean) => void;
 }
 
-export function PlanSheet(props: IProps) {
+export function PlanSheet(this: any, props: IProps) {
```
Comment (gtryus) [inline src/components/PlanSheet.tsx]: I am not sure what you are trying to do here. This doesn't seem to fit the normally calling pattern for a function.

File: src/components/PlanSheet.tsx [hunk 2]
```diff
@@ -299,6 +301,10 @@ export function PlanSheet(props: IProps) {
   const isNum = (value: string | number) =>
     isNumber(value) || /^[0-9]$/.test(value);
 
+  useEffect(() => {
+    setPlanSave(saveRef);
```
Comment (gtryus) [inline src/components/PlanSheet.tsx]: So I think the strategy is to save the dom ref to the save button in a global variable, right? This seems like an unusual strategy. There may be a number of edge cases where the screen may or may not have the save button on it... So the PR may require some more testing before I approve it.

File: src/components/ScriptureTable.tsx [hunk 3]
```diff
@@ -255,9 +255,11 @@ export function ScriptureTable(props: IProps) {
...
           });
         }
-        memory.update((t: TransformBuilder) =>
-          t.removeRecord({ type: id.type, id: id.id })
-        );
+        if (id.id !== '') {
+          memory.update((t: TransformBuilder) =>
+            t.removeRecord({ type: id.type, id: id.id })
+          );
+        }
```
Comment (gtryus) [inline src/components/ScriptureTable.tsx]: This change seems to be the main change referred to by TT-1012. Are the other changes related but addressing different issues?

File: src/global.d.ts [hunk 4]
```diff
@@ -21,6 +21,8 @@ declare module 'reactn/default' {
     bucket: Bucket;
     remote: JSONAPISource;
     remoteBusy: boolean;
+    planSave: React.MutableRefObject;
     open: boolean; // open a dialog
+    planSave: React.MutableRefObject;
```
Comment (gtryus) [inline src/global.d.ts]: Maybe this is a github thing but having the planSave item twice looks like a mistake.

File: src/routes/drawer.tsx [hunk 5]
```diff
@@ -392,15 +393,17 @@ export function ResponsiveDrawer(props: IProps) {
...
     if (saveConfirm.current) saveConfirm.current();
     saveConfirm.current = undefined;
     setAlertOpen(false);
     setChanged(false);
   };
-  const handleUnsaveRefused = () => {
-    saveConfirm.current = undefined;
+  const handleSaveFirst = () => {
     setAlertOpen(false);
+    if (planSave.current) planSave.current.click();
```
Comment (gtryus) [inline src/routes/drawer.tsx]: Where is planSave.current set to undefined?

**Comment by sarahentzel** [comment]
Yes, this is a mess.  I thought these other changes had already gone into develop and it was just a merge/git confusion on my part.  I also thought that I had come around to just changing the question to be more clear and not doing anything more complicated.  Maybe that's what had been merged already and these were leftover changes from trying other changes.  In any case, that one little change in ScriptureTable is all that should be merged.

**Pull request merged**